### ISSUE TITLE
代码问题

### ISSUE THREAD

**Issue by KeanuWongves** [lartpang/ZoomNet#2]
想请问一下在HMU模块上，文章中描述的是每个group的第一个去跟下一个层融合，而对应代码里取的是outs[group_id-1][1]，但是这里的应该为0才跟原文的意思想对应吧。
for group_id in range(1, self.num_groups - 1):
            branch_out = self.interact[str(group_id)](torch.cat([xs[group_id], outs[group_id - 1][1]], dim=1))
            outs.append(branch_out.chunk(3, dim=1))
以及包括后面这两部分的内容， o[0]这里的0是否应该为1才对呢？
        out = torch.cat([o[0] for o in outs], dim=1)
        gate = self.gate_genator(torch.cat([o[-1] for o in outs], dim=1))

**Comment by lartpang** [comment]
@q671383789 感谢你的细心。确实略有出入。

每个`branch_out`分成三部分：
- 代码中：0用于作为各组直接输出，1用于分组之间交互，2用于生成gate值。
- 论文Figure4中：0用于分组之间的交互，1用于生成gate值，2用于各组直接输出。

![image](https://user-images.githubusercontent.com/26847524/161942733-ca8813a6-127a-4324-a834-4da3dc994017.png)

**Comment by KeanuWongves** [comment]
嗯嗯，但这个其实应该是不影响的。还想请教一下作者一个问题就是文中的几个评价指标具体是什么呢，因为不是做COD方向的，看的有点不明白




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年4月6日(星期三) 下午5:23
收件人: ***@***.***&gt;; 
抄送: "系要听Rock &amp; ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [lartpang/ZoomNet] 代码问题 (Issue #2)





 
@q671383789 感谢你的细心。确实略有出入。
 
每个branch_out分成三部分：
  
代码中：0用于作为各组直接输出，1用于分组之间交互，2用于生成gate值。
 
论文Figure4中：0用于分组之间的交互，1用于生成gate值，2用于各组直接输出。

**Comment by lartpang** [comment]
@q671383789 

我不明白你不清楚的是指标的定义还是具体缩写对应的指标计算形式？

**Comment by KeanuWongves** [comment]
指标的定义就可以啦，还想请问一下UAL loss的lamda常数值有经验值推荐吗

**Comment by lartpang** [comment]
@q671383789

由于涉及到指标比较多，建议你直接看看显著性目标检测的最新综述。他们都介绍的很详细。
我只做过lambda=1的实验。其余的可能需要你进一步尝试了。就我论文中的实验来看，余弦形式可能会相对好一些。但是=1实际上也不会差太多。

**Comment by KeanuWongves** [comment]
十分感谢！



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年5月6日(星期五) 下午4:08
收件人: ***@***.***&gt;; 
抄送: "系要听Rock &amp; ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [lartpang/ZoomNet] 代码问题 (Issue #2)





 
@lartpang 由于涉及到指标比较多，建议你直接看看显著性目标检测的最新综述。他们都介绍的很详细。
 我只做过lambda=1的实验。其余的可能需要你进一步尝试了。就我论文中的实验来看，余弦形式可能会相对好一些。但是=1实际上也不会差太多。

**Comment by lartpang** [comment]
那我就先关闭这个问题了

**Comment by KeanuWongves** [comment]
具体是叫什么就可以了，含义我可以自己去搜搜，要问您那也太麻烦了。还有就是增加的额外损失常数值的时候大概经验值是多少，好像文中没有直接说
------------------ 原始邮件 ------------------
发件人: Pang ***@***.***&gt;
发送时间: 2022年4月7日 10:02
收件人: lartpang/ZoomNet ***@***.***&gt;
抄送: q671383789 ***@***.***&gt;, Mention ***@***.***&gt;
主题: Re: [lartpang/ZoomNet] 代码问题 (Issue #2)





 
@q671383789
 
我不明白你不清楚的是指标的定义还是具体缩写对应的指标计算形式？

**Comment by lartpang** [comment]
@q671383789 

关于指标的名字，论文中有写哦：
![image](https://user-images.githubusercontent.com/26847524/195079508-ef74512a-e833-47a7-8bb2-29cd7f08b7d1.png)

唯一可能会造成误解的地方是fmeasure和emeasure选择的形式，我们列出的是maxEmeasure和meanFmeasure，具体的计算方式可以参考我的指标计算库：

https://github.com/lartpang/PySODMetrics

关于损失使用常数值的形式，受限于时间和资源，我们仅仅测试了1的时候。
